### PR TITLE
Role based access control

### PR DESCRIPTION
**Pull request by ngmachado**
This PR will do:

- Implement role based access in bank contract
- Add sugar function to easy add and remove address from roles
- Indexed addresses in events
- Test RAC

### Some notes

Bank owner is now at role `admin`, the bank can have more admins. Only `admin` can add and remove addresses to roles.
Dedicated role to external `keepers` to make liquidations. Liquidations can also be perform by `admin` role users.
Dedicated role to external `price updaters`. Price updates can also be perform by `admin` role users.

`_bankFactoryOwner` is not part of role system. Don't feel that it should be a role based function.

### CODE REVIEW

File: contracts/Bank.sol [hunk 1]
```diff
@@ -325,4 +337,38 @@ contract Bank is BankStorage, Ownable, Initializable {
...
     }
+
+    /**
+     * @dev Allows admin to add address to keeper role
+     * @param keeper address of new keeper
+     */
+    function addKeeper(address keeper) external {
+        require(keeper != address(0), "operation not allowed");
+        grantRole(KEEPER_ROLE, keeper);
+    }
```
Comment (mikeghen) [inline contracts/Bank.sol:348]: Is a require or modify needed here? Is this only ADMIN_ROLE?

File: contracts/Bank.sol [hunk 2]
```diff
@@ -325,4 +337,38 @@ contract Bank is BankStorage, Ownable, Initializable {
...
+        require(updater != address(0), "operation not allowed");
+        grantRole(PRICE_UPDATER_ROLE, updater);
+    }
+
+    /**
+     * @dev Allows admin to remove address from price updater role
+     * @param oldUpdater address of old price updater
+     */
+    function revokePriceUpdater(address oldUpdater) external {
+        revokeRole(PRICE_UPDATER_ROLE, oldUpdater);
```
Comment (mikeghen) [inline contracts/Bank.sol:372]: Can you change `PRICE_UPDATER` to `REPORTER`, sticking with Tellor terminology for that role

File: contracts/BankStorage.sol [hunk 3]
```diff
@@ -11,6 +11,10 @@ import "@openzeppelin/contracts/access/Ownable.sol";
 contract BankStorage {
     /*Variables*/
     string name;
+    // role identifier for keeper that can make liquidations
+    bytes32 public constant KEEPER_ROLE = keccak256("KEEPER_ROLE");
+    // role identifier for price updater
+    bytes32 public constant PRICE_UPDATER_ROLE = keccak256("PRICE_UPDATER_ROLE");
```
Comment (mikeghen) [inline contracts/BankStorage.sol:17]: Use `REPORTER_ROLE`

**Review by mikeghen — CHANGES_REQUESTED** [review]

File: contracts/BankStorage.sol (hunk 3 above)
Comment (ngmachado) [reply contracts/BankStorage.sol]: done

File: contracts/Bank.sol (hunk 1 above)
Comment (ngmachado) [reply contracts/Bank.sol:348]: is not needed, implementation take care of checking. Added a test for completion on this question

**Review by mikeghen — APPROVED** [review]

**Pull request merged**